test/compilable/test19954.d

```diff
@@ -0,0 +1,6 @@
+// https://issues.dlang.org/show_bug.cgi?id=19954
+
+void main()
+{
+    auto a = cast(char[][]) "";
```

Comment:
Should this really compile? `auto a = cast(int[]) 1;` doesn't.

Reply:
According to [this](https://run.dlang.io/is/FIELFW) (run "All dmd com"), yes.  But I think `""` is more analogous to `[]`, than to `1`.

Reply:
Ah, a regression. `cast(void[][]) []` does compile too, so all good.